### PR TITLE
Modify the uploader.js in routes

### PR DESCRIPTION
**Pull request by r05921104**
Receive the post request from the front end
1. If the posted tagkey is already in the database, there would be an unique violation error, which is represented by "23505" error code and the "辨識碼已被使用過" message would be sent back to the front end.
2. Otherwise, the "成功上傳"message would be sent back.

### CODE REVIEW

File: routes/uploader.js [hunk 1]
```diff
@@ -20,7 +20,50 @@ var storage = multer.diskStorage({
...
+        else{
+          console.log('Complete');
+          res.json({
+            state: 'success',
+            state_array: item_state_array
+          });
+        }
+      }
+    });
+  }
```
Comment (tz70s) [inline routes/uploader.js]: To be style consistent, it's better write 
```javascript
if (condition) {

} else {

}
```

Comment (r05921104) [reply routes/uploader.js]: ok, I would modify it in this style later.

File: routes/uploader.js [hunk 2]
```diff
@@ -20,7 +20,50 @@ var storage = multer.diskStorage({
...
+    var datecode = req.body['datecode'+i];
+    var quantity = req.body['quantity'+i];
+    var sql = "insert into goods (tagkey, orderid, itemid, datecode, quantity) values ('" + tagkey +"', '" + orderid +"', '" + itemid +"', '" + datecode +"', '" + quantity +"')";
+    dbclient.query(sql, function(err, result){
+      if(err){
+        if (err['code'] === '23505') {//Unique violation
+          item_state_array.push({'upload_state':'辨識碼已被使用過'});
+          complete_number++;
+          if(complete_number < total_number){
+            console.log('Not complete');
```
Comment (tz70s) [inline routes/uploader.js]: Better to use logger which added in `utils/log.js`
```javascript
var logger = require('../utils/log.js')

// Replace console.log
logger.log("Not complete");

```

File: static/js/uploader.js [hunk 3]
```diff
@@ -6,77 +6,124 @@ $(document).ready(function() {
...
-              success: function(data) {
-                console.log('success');
-              }
-      });
+            type: 'POST',
+            url: window.location.protocol + '//' + window.location.host + '/uploader',
+            data: $('#upload_order_form').serialize(),
+            dataType: 'json',
+            success: function(data) {
+              //console.log(data['state_array']);
```
Comment (tz70s) [inline static/js/uploader.js]: Remove this.

File: static/js/uploader.js [hunk 4]
```diff
@@ -6,77 +6,124 @@ $(document).ready(function() {
...
+              var state_array = data['state_array'];
+              for(var i=1;i<=state_array.length;i++){
+                $('#upload_status'+i).val(state_array[i-1]['upload_state']);
+            }
```
Comment (tz70s) [inline static/js/uploader.js]: How about using index from 0 instead of 1?

**Comment by tz70s** [comment]
Please squash these commits, I'll use additional tool to fix these up.
To avoid conflict, merge this first.

**Comment by tz70s** [comment]
Great thanks!

**Pull request merged**